target-chains/cosmwasm/tools/src/deployer/index.ts

```diff
@@ -0,0 +1,32 @@
+import { CONFIG, CONFIG_TYPE } from "./config";
+import { TerraDeployer } from "./terra";
+import { InjectiveDeployer } from "./injective";
+import { NETWORKS } from "../network";
+
+export interface Deployer {
+  deployArtifact(artifact: string): Promise<number>;
+  instantiate(
+    codeId: number,
+    inst_msg: string | object,
+    label: string
+  ): Promise<string>;
+  migrate(contract: string, codeId: number): Promise<void>;
+}
+
+export class DeployerFactory {
+  static create(network: NETWORKS, mnemonic: string): Deployer {
```

Comment:
this function would be more flexible if it took the `NetworkConfig` as an argument. That would allow callers to create a custom config instead of forcing them to use a config from `CONFIG`

Reply:
yeah makes sense